src/libostree/ostree-repo-pull.c

```diff
@@ -3619,6 +3789,7 @@ ostree_repo_pull_with_options (OstreeRepo             *self,
 
   if (pull_data->legacy_transaction_resuming)
     g_debug ("resuming legacy transaction");
+  pull_data->requested_refs_to_fetch = requested_refs_to_fetch;
```

Comment:
I was thinking about it, but `pull_data` will not outlive the `requested_refs_to_fetch` variable. I can add it, for good measure.